tools/sdk-generation-pipeline/packages/sdk-generation-cli/src/utils/autorestConfigExtractorUtils.ts

```diff
@@ -0,0 +1,39 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import winston from 'winston';
+
+export function extractAutorestConfigs(autorestConfigFilePath: string, sdkRepo: string, logger: winston.Logger): string | undefined {
+    if (!fs.existsSync(autorestConfigFilePath)) {
+        return undefined;
+    }
+    const autorestConfigFileContent = fs.readFileSync(autorestConfigFilePath, 'utf-8');
+    try {
+        let autorestConfigs: string[] = autorestConfigFileContent.split(/#+ *azure-sdk-for-/);
+        if (autorestConfigs.length < 2) {
+            throw new Error(`Parse autorest config file failed.`);
+        }
+        autorestConfigs = autorestConfigs.slice(1);
+        if (!path.basename(sdkRepo).startsWith('azure-sdk-for-')) {
```

Comment:
What's sdkRepo? is it the sdkRepo directory path? It is strange that assign sdkRepo to 'azure-sdk-for-'

Reply:
When docker container is running in pipeline, sdk repo is mounted in docker container as `/sdkRepo`. We don't assign sdkRepo to `azure-sdk-for`. It's just used for identitying whether the docker is ruinning in pipeline.